website/static/js/share/utils.js

```diff
@@ -190,30 +228,269 @@ utils.removeFilter = function(vm, filter){
     utils.search(vm);
 };
 
-utils.arrayEqual = function(a, b) {
+/* Tests array equality */
+utils.arrayEqual = function (a, b) {
     return $(a).not(b).length === 0 && $(b).not(a).length === 0;
 };
 
-utils.loadStats = function(vm){
-    vm.statsLoaded(false);
+utils.addFiltersToQuery = function (query, filters) {
+    if (filters) {
+        filters.forEach(function (filter) {
+            query = utils.filteredQuery(query, filter.filter);
+        });
+    }
+    return query;
+};
 
+/* Loads the raw and normalized data for a specific result */
+utils.loadRawNormalized = function(result){
+    var nonJsonErrors = function(xhr) {
+        return xhr.status > 200 ? JSON.stringify(xhr.responseText) : xhr.responseText;
+    };
+    var source = encodeURIComponent(result.shareProperties.source);
+    var docID = encodeURIComponent(result.shareProperties.docID);
     return m.request({
         method: 'GET',
-        url: '/api/v1/share/stats/?' + $.param({q: utils.buildQuery(vm)}),
-        background: true
+        url: '/api/v1/share/documents/' + source + '/' + docID + '/',
+        extract: nonJsonErrors
     }).then(function(data) {
-        vm.statsData = data;
-        $.map(Object.keys(vm.graphs), function(type) {
-            if(type === 'shareDonutGraph') {
-                var count = data.charts.shareDonutGraph.columns.filter(function(val){return val[1] > 0;}).length;
-                $('.c3-chart-arcs-title').text(count + ' Provider' + (count !== 1 ? 's' : ''));
+
+        var normed = JSON.parse(data.normalized);
+        normed = JSON.stringify(normed, undefined, 2);
+
+        var all_raw = JSON.parse(data.raw);
+        result.raw = all_raw.doc;
```

Comment:
but at some point, we do want to connect this response to the result (since what is returned is the raw/normalized document for that result). At least I think, @erinspace, is that correct?